### DISCUSSION TITLE
How is this library built?

### DISCUSSION THREAD

**Discussion by strogonoff** [ai/nanoid#430] [General]
I noticed that this library switched to ESM not long ago, and its setup seems to be fairly elegant from what I can see, so I became curious.

However, I haven’t noticed anything that would handle the build/publish automation: no scripts, bundler configurations, GHA workflows or a Makefile. I have also noticed no README documentation that would explain the process of build & release.

If I missed something obvious… please feel free to close! But if not, how is this library built (if at all) to be available for most users?

I’m not sure if this is just a discussion or a documentation-related issue. As someone who would be vetting this library as a new dependency, things like this would play a role in my decision, and signal that there is some established process that will continue even if original maintainer becomes incapacitated or bored with the library.

Thanks for great work!

**Comment by ai** [comment]
> However, I haven’t noticed anything that would handle the build/publish automation: no scripts, bundler configurations, GHA workflows or a Makefile. I have also noticed no README documentation that would explain the process of build & release.

Yes. The build step increases maintainability cost and complexity of the library.

I don't use build step in all my tools.

I used build step before for Autoprefixer and PostCSS and my decision to avoid build step is the result of my previous practical experience.

The only “build step” is using [`clean-publish`](https://github.com/shashkovdanil/clean-publish) instead of `npm publish`. It is just clean unnecessary fields in `package.json` and long comments in JS sources.

**Reply by strogonoff** [reply]
Thanks! But I noticed that package.json on NPM is different from package.json in this repository. For example, there is no `scripts`. Am I being mistaken, or does it mean there is a sort of build process only done by hand?

**Reply by ai** [reply]
It was done by that `clean-publish`
